walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -28,11 +28,27 @@
       return new ArrayList<TimeRange>();
     }
     
-    List<TimeRange> unavailableTimes = getSortedUnavailableTimes(events, request);
-    return computeAvailableTimes(unavailableTimes, request);
+    // The Collection that stores the available times for the requested meeting.
+    Collection<TimeRange> availableTimes;
+
+    List<TimeRange> unavailableTimesWithOptionals = getSortedUnavailableTimes(events, request, true);
+    Collection<TimeRange> timesWithOptionals = computeAvailableTimes(unavailableTimesWithOptionals, request);
+
+    List<TimeRange> unavailableTimesWithoutOptionals = getSortedUnavailableTimes(events, request, false);
+    Collection<TimeRange> timesWithoutOptionals = computeAvailableTimes(unavailableTimesWithoutOptionals, request);
+
+    if (timesWithOptionals.isEmpty() && (request.getAttendees().size() == 0)) {
+      availableTimes = timesWithOptionals;
+    } else if (timesWithOptionals.isEmpty() && (request.getAttendees().size() != 0)) {
+      availableTimes = timesWithoutOptionals;
+    } else {
+      availableTimes = timesWithOptionals;
+    }
+
+    return availableTimes;
```

Comment:
Copying & pasting what I wrote to Alvin:
Please add a good comment in the method header on what it returns ("if there is an opening for eveyone, return it, otherwise return the opening for the required").
In general, this is not the most recommended interface design, because the caller wouldn't (explicitly) know whether the returned openings are for everyone or for the required only. It is recommended to make the interface explicit about what it returns to minimize any potential error/bug.
Having said that, let's not change the interface but add an explicit comments (it's not worth spending time for a new interface).

Reply:
Ack. I've added a comment in the method header now that explains what the query() function does